File: PPScannerApp/PPScannerApp/ViewController.m
//
//  ViewController.m
//  PPScannerApp
//
//  Created by Jura on 16/02/16.
//  Copyright © 2016 MicroBlink. All rights reserved.
//

#import "ViewController.h"

#import <PPScanner/PPScannerFramework.h>

@interface ViewController () <PPScannerDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)viewDidAppear:(BOOL)animated {
    PPScanner *scanner = [[PPScanner alloc] init];

    UIViewController<PPScannerController> *controller = [scanner controller];

    [scanner setDelegate:self];

    [controller update];

    [scanner run];
}

- (void)didFinish {
    NSLog(@"Did finish");
}

- (void)didStart {
    NSLog(@"Did start");
}

@end

File: iOSSimpleSample/iOSSimpleSample/ViewController.cs
using System;
using iOSSimpleBinding;

using UIKit;
using System.Diagnostics;

namespace iOSSimpleSample
{
	public partial class ViewController : UIViewController
	{
		public ViewController (IntPtr handle) : base (handle)
		{
		}

		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();
			// Perform any additional setup after loading the view, typically from a nib.

			PPScanner p = new PPScanner ();

			PPScannerController controller = p.Controller;

			PPScannerDelegate pd = new MyDelegate ();

			p.SetDelegate (pd);

			p.Run ();

		}

		public override void DidReceiveMemoryWarning ()
		{
			base.DidReceiveMemoryWarning ();
			// Release any cached data, images, etc that aren't in use.
		}

		public class MyDelegate : PPScannerDelegate {

			public override void DidStart ()
			{
				Debug.WriteLine ("I am here");
			}


			#region implemented abstract members of PPScannerDelegate
			public override void DidFinish ()
			{
				throw new NotImplementedException ();
			}
			#endregion
		}

	}
}

